### ISSUE TITLE
accessibility all-in-one tree view

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-ui#993]
> Discussed at a Cylc conference a long time back (maybe 2019?) but never written up.

The Cylc 8 UI heavily depends on graphics to convey information, this is intentional. We support changing font-size and colour theme to support minor visual imparements, however, the UI will be challenging for some (e.g. has not support at all for total blindness). We are not currently aware of a demand for an interface which does not depend on graphics amongst the current user base, however, we do have a plan to implement one if the demand materialises.

This would be a simplified representation rather than a "cut-down" version with all the same data accessible but without the panes, tabs, icons and menus of the regular interface. 

It would be implemented as a special "route" (i.e. a separate page with a special URL) which represents everything including GScan, the Tree view and the app itself in a single tree:

* Cylc UI
  * settings
  * workflows
    * a/ (workflow path)
      * run1 (workflow)  -- !spring node!
        * 1 (cycle)
          * foo (task)
            * 01 (job)
         * 2 (cycle)
           * 01 (job)

This approach relies on a hierarchical DoM which works well for screen readers as it has a logical tab order and simple controls at each node (up, down, left, right). Each node would have simple metadata i.e. the node type (workflow, cycle, task, etc) which would be displayed in plain text (not as icons) with supplemental information (metadata, platform, etc) would be "field separated" to aid digestibility. Interactivity would be served via a dedicated link/button (or possibly tree node). This UI can be scaled to any font-size and, navigated by sound with no need for graphics or colour.

The data provision would be shared with the current UI, this would be a separate view on the same data. It's simple design should be easy to implement (the original tree view worked in a similar way).

Challenges:

* Spring nodes
  * We can't expand the *whole* tree when the UI is loaded (as this would mean listing all jobs for all workflows which would be too much to display.
  * So we would have to list cycles/task/jobs only when the workflow node is expanded (as intended for Tui).
  * This is fine, however, would cause the UI to become "jumpy" as bits are being loaded after the page was rendered which is a no-no for accessibility.
  * Because we are doing this within a tree structure the additional data doesn't impact the tree layout so I think it would be OK but would need to investigate.
  * Otherwise would need to separate the GScan and workflow Tree views  so they can be navigated between in the conventional manner.
* Interactivity.
  * Same issue as with spring nodes, we need to list mutations without making the page jumpy.
* Screen reader integration (namely ARIA, not a big challenge but a reasonable learning exercise).
* Ideally would "capture" screen readers at the app entry point and offer them redirection to this view.

**Pull requests welcome!**
